ruoccofabrizio/azure-open-ai-embeddings-qna
# This PR introduces support for PGVector as a feature in the vectorstore, and extends functionality to Azure China and Azure Global for both PGVector and Redis Stack.

## Changes Made
1. **PGVector Support:**
   - Added PGVector as a supported feature in the vectorstore.
   - Implemented necessary changes to ensure compatibility and proper integration.

2. **Azure China and Azure Global Support:**
   - Extended support for PGVector to Azure China and Azure Global environments.
   - Added functionality to enable PGVector and Redis Stack in these environments.

## Motivation and Context
This PR addresses the need for PGVector support in the vectorstore and the requirement to expand compatibility to Azure China and Azure Global. The inclusion of PGVector enhances the capabilities of the vectorstore, and the extended support for Azure regions allows for broader usage.

## How to Test
1. Deploy the modified code to a test environment.
2. Verify that PGVector is functioning as expected in the vectorstore.
3. Test the functionality of PGVector and Redis Stack in both Azure China and Azure Global environments.

## Screenshots
![pg](https://github.com/ruoccofabrizio/azure-open-ai-embeddings-qna/assets/6263170/b543059b-b759-4932-9e17-c497944a0c7f)
![pg2](https://github.com/ruoccofabrizio/azure-open-ai-embeddings-qna/assets/6263170/ebafed9d-36fc-4a99-a9c0-f0a7529cfd38)
![pg3](https://github.com/ruoccofabrizio/azure-open-ai-embeddings-qna/assets/6263170/5196276e-d421-43eb-8980-5611004c1e7d)
![pgcn](https://github.com/ruoccofabrizio/azure-open-ai-embeddings-qna/assets/6263170/08ba6f89-0d59-4f81-a6c3-efe2e2800683)
![pgcn2](https://github.com/ruoccofabrizio/azure-open-ai-embeddings-qna/assets/6263170/3a7cc5b3-59a4-4a21-ba77-4abe6f121d6f)
![pgcn3](https://github.com/ruoccofabrizio/azure-open-ai-embeddings-qna/assets/6263170/6d62a9bb-15d4-4478-ab22-1f248ce4ee98)
![redis](https://github.com/ruoccofabrizio/azure-open-ai-embeddings-qna/assets/6263170/6abc770a-611b-4b98-8cef-cbb9f7ac389f)
![redis_cn](https://github.com/ruoccofabrizio/azure-open-ai-embeddings-qna/assets/6263170/7e34f659-4a2c-4a17-bc6a-56fbfa38bb5f)
![redis_cn2](https://github.com/ruoccofabrizio/azure-open-ai-embeddings-qna/assets/6263170/68ba727f-5d9b-4287-a0e9-782112fd18a5)
![redis2](https://github.com/ruoccofabrizio/azure-open-ai-embeddings-qna/assets/6263170/dea2b2ee-81ea-4d5d-8853-e69964e15c8b)
